typescript/channel.ts

```diff
@@ -132,10 +139,15 @@ export class Channel {
         }
         if (msg.ID === codec.EofID) {
             this.gotEOF = true;
-            // if (this.readers.length > 0) {
-            // 	this.readers.shift()();
-            // }
-            return;
+            // TODO does this block if there are a lot of readers pending?
+            // is there a better way to interleave those?
+            while (true) {
+                let reader = this.readers.shift();
+                if (reader === undefined) {
+                    return;
+                }
+                reader();
+            }
```

Comment:
@progrium so do you have a preference on what this should do if there are multiple readers pending?
`shutdown()` includes a loop over the pending readers, but it seems like if we get an EOF we should also flush those pending reads.

Reply:
I guess there are some similar questions for `handleData` too. Right now it will call the first reader, but there's a chance that the first read will leave some additional data in the buffer that could be handled by subsequent pending reads. So, should that continue flushing any read requests until the buffer is empty?

Reply:
yea EOF should flush